asciidoc/cypher-workflow.adoc

```diff
@@ -0,0 +1,804 @@
+Since read servers are typically more plentiful than write servers, it is beneficial to direct read traffic to read servers instead of the write server.
+Doing so helps in keeping write servers available for write transactions.
+
+Access mode is generally specified by the method used to call the transaction function.
+Session classes provide a method for calling reads and another for writes.
+
+As a fallback for auto-commit and unmanaged transactions, a default access mode can also be provided at session level.
+This is only used in cases when the access mode cannot otherwise be specified, and will be overridden should a transaction function be used within that session.
+
+[NOTE]
+The driver does not parse Cypher and therefore cannot automatically determine whether a transaction is intended to carry out read or write operations.
+As a result, a write transaction tagged as a read will still be sent to a read server, but will fail on execution.
+
+[.tabbed-example]
+.Read-write transaction
+====
+[.include-with-dotnet]
+======
+[source, csharp]
+----
+include::{dotnet-examples}/Examples.cs[tags=read-write-transaction]
+----
+======
+
+// [.include-with-go]
+// ======
+// [source, go]
+// ----
+// include::{go-examples}/examples_test.go[tags=read-write-transaction]
+// ----
+// ======
+
+[.include-with-java]
+======
+[source, java]
+----
+include::{java-examples}/ReadWriteTransactionExample.java[tags=read-write-transaction]
+----
+======
+
+[.include-with-javascript]
+======
+[source, javascript]
+----
+include::{javascript-examples}/examples.test.js[tags=read-write-transaction]
+----
+======
+
+// [.include-with-python]
+// ======
+// [source, python]
+// ----
+// include::{python-examples}/read_write_transaction_example.py[tags=read-write-transaction]
+// ----
+// ======
+====
+
+[[driver-databases]]
+== Databases and execution context
+
+Neo4j 4.0 introduces the ability to work with multiple databases within the same DBMS.
+For a single instance, this is limited to one user database, plus the `system` database.
+From a driver API perspective, a database can be selected on session construction, and is used as an execution context for the transactions within that session.
+It is not currently possible to execute a transaction across multiple databases.
+
+In a multi-database environment, the server tags one database as default.
+This is selected whenever a session is created without naming a particular database as default.
+
+In an environment with only a single database, that database is always the default.
+
+For more information about the multi-database feature, refer to the <<cypher-manual#cypher-manual, Cypher Manual>> which has a full breakdown of the Neo4j data storage hierarchy.
+
+=== The `system` database
+
+Some administrative operations must be carried out against the `system` database.
+To do this, construct a session targeting the database called `system` and execute Cypher as usual.
```

Comment:
Perhaps give an example of what "some administrative operations" are, and/or link to the relevant sections in Cypher Manual?

Reply:
I'm not sure exactly, and in particular not sure what is exposed in Community vs Enterprise.

Reply:
Added link to Admin section and Databases in particular, of Cypher Manual.